server/app/uk/gov/ons/addressIndex/server/controllers/AddressController.scala

```diff
@@ -38,19 +44,45 @@ class AddressController @Inject()(
     * @param format requested format of the query (paf/nag)
     * @return Json response with addresses information
     */
-  def addressQuery(input: String, format: String): Action[AnyContent] = Action async { implicit req =>
-    logger info s"#addressQuery:\ninput $input , format: $format"
-    input.toOption map { actualInput =>
-      val tokens = parser tag actualInput
-      logger info s"#addressQuery parsed:\n${tokens.map(t => s"value: ${t.value} , label:${t.label}").mkString("\n")}"
-      formatQuery[AddressBySearchResponseContainer, Seq[CrfTokenResult]](
-        formatStr = format,
-        inputForPafFn = AddressQueryInput(tokens),
-        pafFn = pafSearch,
-        inputForNagFn = AddressQueryInput(tokens),
-        nagFn = nagSearch
-      ) getOrElse futureJsonBadRequest(UnsupportedFormat)
-    } getOrElse futureJsonBadRequest(EmptySearch)
+  def addressQuery(input: String, format: String, offset: Option[String] = None, limit: Option[String] = None): Action[AnyContent] = Action async { implicit req =>
```

Comment:
They shouldn't be. 
Checkout using `Option[Int]` with a default in the routes file.
Heres the Play Framework docs: 
https://www.playframework.com/documentation/2.5.x/ScalaRouting

more docs:
https://www.playframework.com/documentation/1.2.4/routes

Other refs:
http://mariussoutier.com/blog/2012/12/10/playframework-routes-part-1-basics/